modules/dynamic-sampling-config/src/main/scala/io/janstenpickle/trace4cats/sampling/dynamic/config/ConfiguredHotSwappableDynamicSpanSampler.scala

```diff
@@ -0,0 +1,32 @@
+package io.janstenpickle.trace4cats.sampling.dynamic.config
+
+import cats.effect.kernel.{Resource, Temporal}
+import io.janstenpickle.trace4cats.kernel.SpanSampler
+import io.janstenpickle.trace4cats.model.{SampleDecision, SpanContext, SpanKind, TraceId}
+import io.janstenpickle.trace4cats.sampling.dynamic.HotSwappableDynamicSpanSampler
+
+trait ConfiguredHotSwappableDynamicSpanSampler[F[_]] extends SpanSampler[F] {
+  def updateConfig(config: SamplerConfig): F[Boolean]
+  def getConfig: F[SamplerConfig]
+}
+
+object ConfiguredHotSwappableDynamicSpanSampler {
```

Comment:
I'm not sure this is the best name for this 😂

Reply:
@janstenpickle Looking at this I can taste good old Java in my mouth...

Reply:
I think it needs a few more `Proxy` and `Factory` in the name